test/api/controllers/validators/ValidatorSpec.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.controllers.validators
+
+import api.controllers.validators.resolvers.{ResolveJsonObject, ResolveNino, ResolveTaxYear}
+import api.models.domain.{Nino, TaxYear}
+import api.models.errors._
+import cats.data.Validated
+import cats.data.Validated.Invalid
+import cats.implicits._
+import org.scalamock.scalatest.MockFactory
+import play.api.http.Status.BAD_REQUEST
+import play.api.libs.json.{JsValue, Json, Reads}
+import support.UnitSpec
+
+class ValidatorSpec extends UnitSpec with MockFactory {
+
+  private implicit val correlationId: String = "1234"
+
+  private val validNino    = Nino("AA123456A")
+  private val validTaxYear = TaxYear.fromMtd("2023-24")
+
```

Comment:
Is this file formatted?

Reply:
Yes, it is formatted. I tried it again :)